### ISSUE TITLE
Unable to create data disk of provisioning type as Fat

### ISSUE THREAD

**Issue by kiranchavala** [apache/cloudstack#8644]
ISSUE TYPE

BUG 

Component Name

Storage

CLOUDSTACK VERSION

Cloudstack version 4.18, 4.19 

SUMMARY

Unable to create data disk of provisioning type as Fat

Steps to reproduce the issue

1. Create a disk offering with Provisioning type as fat and storage type  as local

![Screenshot 2024-02-13 at 10 07 11 AM](https://github.com/apache/cloudstack/assets/1401014/6c31a6c3-2fbe-4737-8256-f58e908079f1)

2. Create a data disk based on the disk offering and attach it to a  running vm

3. Check the data disk size using "qemu-img" command 

The actual disk size is thin instead of thick

```

[root@ol8 agent]# qemu-img info /localstorage/655c8f16-1574-420f-bfd1-adbf8e7ca90d
image: /localstorage/655c8f16-1574-420f-bfd1-adbf8e7ca90d
file format: qcow2
virtual size: 10 GiB (10737418240 bytes)
disk size: 196 KiB
cluster_size: 65536
Format specific information:
    compat: 1.1
    compression type: zlib
    lazy refcounts: false
    refcount bits: 16
    corrupt: false
    extended l2: false

```

Logs 

```
2024-02-12 12:30:04,757 DEBUG [cloud.agent.Agent] (agentRequest-Handler-5:null) (logid:1637c086) Request:Seq 1-7785597855816746741:  { Cmd , MgmtId: 32988167406101, via: 1, Ver: v1, Flags: 100011, [{"org.apache.cloudstack.storage.command.CreateObjectCommand":{"data":{"org.apache.cloudstack.storage.to.VolumeObjectTO":{"uuid":"655c8f16-1574-420f-bfd1-adbf8e7ca90d","volumeType":"DATADISK","dataStore":{"org.apache.cloudstack.storage.to.PrimaryDataStoreTO":{"uuid":"2284dd6f-8bc7-431e-afa8-e247bfe1e940","name":"ol8.localdomain-local-2284dd6f","id":"1","poolType":"Filesystem","host":"10.0.35.197","path":"/localstorage","port":"0","url":"Filesystem://10.0.35.197/localstorage/?ROLE=Primary&STOREUUID=2284dd6f-8bc7-431e-afa8-e247bfe1e940","isManaged":"false"}},"name":"test-kiran-local","size":"(10.00 GB) 10737418240","volumeId":"21","accountId":"2","provisioningType":"FAT","poolId":"1","id":"21","cacheMode":"NONE","hypervisorType":"KVM","directDownload":"false","deployAsIs":"false"}},"wait":"0","bypassHostMaintenance":"false"}}] }

2024-02-12 12:30:04,757 DEBUG [cloud.agent.Agent] (agentRequest-Handler-5:null) (logid:1637c086) Processing command: org.apache.cloudstack.storage.command.CreateObjectCommand

2024-02-12 12:30:04,757 DEBUG [storage.resource.StorageSubsystemCommandHandlerBase] (agentRequest-Handler-5:null) (logid:1637c086) Executing command CreateObjectCommand: [{"data":{"org.apache.cloudstack.storage.to.VolumeObjectTO":{"uuid":"655c8f16-1574-420f-bfd1-adbf8e7ca90d","volumeType":"DATADISK","dataStore":{"org.apache.cloudstack.storage.to.PrimaryDataStoreTO":{"uuid":"2284dd6f-8bc7-431e-afa8-e247bfe1e940","name":"ol8.localdomain-local-2284dd6f","id":1,"poolType":"Filesystem","host":"10.0.35.197","path":"/localstorage","port":0,"url":"Filesystem://10.0.35.197/localstorage/?ROLE=Primary&STOREUUID=2284dd6f-8bc7-431e-afa8-e247bfe1e940","isManaged":false}},"name":"test-kiran-local","size":10737418240,"volumeId":21,"accountId":2,"provisioningType":"FAT","poolId":1,"id":21,"cacheMode":"NONE","hypervisorType":"KVM","directDownload":false,"deployAsIs":false}},"wait":0,"bypassHostMaintenance":false}].


2024-02-12 12:30:04,761 DEBUG [utils.script.Script] (agentRequest-Handler-5:null) (logid:1637c086) Executing: qemu-img create -f qcow2 /localstorage/655c8f16-1574-420f-bfd1-adbf8e7ca90d 10737418240

**2024-02-12 12:30:04,778 DEBUG [utils.script.Script] (agentRequest-Handler-5:null) (logid:1637c086) Formatting '/localstorage/655c8f16-1574-420f-bfd1-adbf8e7ca90d', fmt=qcow2 cluster_size=65536 extended_l2=off compression_type=zlib size=10737418240 lazy_refcounts=off refcount_bits=16**
```

Actual Behaviour

The size of the data disk is not thick 

Expected Behaviour

The size of the data disk should be thick

**Comment by kiranchavala** [comment]
**The issue is not observed for root disk where the service offering is of  Provisioning type as fat and storage type  as local**


1. Create a service offering Provisioning type as fat and storage type  as local
![Screenshot 2024-02-13 at 10 07 38 AM](https://github.com/apache/cloudstack/assets/1401014/3159b746-e8d4-48f7-97ba-4f6acfabd2cf)



2. Launch a vm based on servcice offering 

3. Check the root disk size using "qemu-img" command 



[root@ol8 agent]# qemu-img info /localstorage/11b9756f-29cf-47ba-a57a-2e764714ccb6
image: /localstorage/11b9756f-29cf-47ba-a57a-2e764714ccb6
file format: qcow2
virtual size: 8 GiB (8589934592 bytes)
disk size: 8 GiB
cluster_size: 65536
Format specific information:
    compat: 1.1
    compression type: zlib
    lazy refcounts: false
    refcount bits: 16
    corrupt: false
    extended l2: false


Logs
----

```
2024-02-12 12:28:07,876 DEBUG [utils.script.Script] (agentRequest-Handler-3:null) (logid:e009fed7) Executing: qemu-img create -f qcow2 /localstorage/11b9756f-29cf-47ba-a57a-2e764714ccb6 8589934592

2024-02-12 12:28:07,878 DEBUG [utils.script.Script] (agentRequest-Handler-3:null) (logid:e009fed7) Executing while with timeout : 3600000
2024-02-12 12:28:07,893 DEBUG [utils.script.Script] (agentRequest-Handler-3:null) (logid:e009fed7) Execution is successful.

2024-02-12 12:28:07,893 DEBUG [utils.script.Script] (agentRequest-Handler-3:null) (logid:e009fed7) Formatting '/localstorage/11b9756f-29cf-47ba-a57a-2e764714ccb6', fmt=qcow2 cluster_size=65536 extended_l2=off compression_type=zlib size=8589934592 lazy_refcounts=off refcount_bits=16

2024-02-12 12:28:07,902 DEBUG [utils.script.Script] (agentRequest-Handler-3:null) (logid:e009fed7) Executing: qemu-img convert -O qcow2 -o preallocation=full -U --image-opts driver=qcow2,file.filename=/localstorage/36397421-b771-4562-827e-6019cec9b101 /localstorage/11b9756f-29cf-47ba-a57a-2e764714ccb6
```

**Comment by kiranchavala** [comment]
The issue is not observed for data disk which is created with Provisioning type as fat and storage type  as shared

1. Create a disk offering with Provisioning type as fat and storage type  as shared

![Screenshot 2024-02-13 at 10 22 00 AM](https://github.com/apache/cloudstack/assets/1401014/2bd54695-96df-4d6f-9486-e02db57f774c)



2. Creat a data disk based on the disk offering and attach it to a vm

3. Check the data disk size using "qemu-img" command 



[root@ol8 572d3541-3854-3123-9590-76639ffcd579]# qemu-img info 49fae15b-ab71-4733-a087-ca429e34d3cc
image: 49fae15b-ab71-4733-a087-ca429e34d3cc
file format: qcow2
virtual size: 10 GiB (10737418240 bytes)
disk size: 11 GiB
cluster_size: 65536
Format specific information:
    compat: 1.1
    compression type: zlib
    lazy refcounts: false
    refcount bits: 16
    corrupt: false
    extended l2: false


Logs
----

2024-02-13 04:39:12,821 DEBUG [utils.script.Script] (agentRequest-Handler-3:null) (logid:28c0fe7b) Executing: qemu-img create -o preallocation=full -f qcow2 /mnt/572d3541-3854-3123-9590-76639ffcd579/49fae15b-ab71-4733-a087-ca429e34d3cc 10737418240

**Comment by rohityadavcloud** [comment]
This maybe hypervisor dependent.

**Comment by DaanHoogland** [comment]
fixed in #9141
